src/middleware/router.js

```diff
@@ -537,6 +556,110 @@ function sendHttpRequest (ctx, route, options) {
   })
 }
 
+// Send secondary route request
+const sendSecondaryRouteHttpRequest = (ctx, route, options) => {
+  return new Promise((resolve, reject) => {
+    const response = {}
+    let { downstream } = ctx.state
+    let method = http
+
+    if (route.secured) {
+      method = https
+    }
+
+    const routeReq = method.request(options)
+      .on('response', routeRes => {
+        response.status = routeRes.statusCode
+        response.headers = routeRes.headers
+
+        if(!bucket) {
+          bucket = getGridFSBucket()
+        }
+
+        const uploadStream = bucket.openUploadStream()
+        response.bodyId = uploadStream.id
+
+        if (!ctx.authorisedChannel.responseBody) {
+          // reset response body id
+          response.bodyId = null
+        }
+
+        uploadStream
+          .on('error', (err) => {
+            logger.error(`Error streaming secondary route response body from '${options.path}' into GridFS: ${err}`)
+          })
+          .on('finish', (file) => {
+            logger.info(`Streamed secondary route response body from '${options.path}' into GridFS, body id ${file._id}`)
+          })
+
+        const responseBuf = []
+        routeRes
+          .on('data', chunk => {
+            if (!response.timestamp) {
+              response.timestamp = new Date()
+            }
+
+            // Storing of the secondary route response is not done when the primary route fails
+            if (ctx.primaryRouteFailure) {
+              uploadStream.abort(() => {
```

Comment:
Im not sure this is working as expected. If the main route is isnt reachable, i can still see the secondary route being sent while the transactions has failed

Reply:
Fixed, now working